app/controllers/ArrivalNotificationController.scala

```diff
@@ -77,23 +79,35 @@ class ArrivalNotificationController @Inject()(
 
                         }
                         .recover {
-                          case _ => InternalServerError
+                          case _ => {
+                            InternalServerError(Json.toJson(ErrorResponseBuilder.failedSavingArrivalNotification))
+                              .as("application/json")
+                          }
                         }
                     }
                     case Left(FailedToValidateXml) =>
-                      Future.successful(BadRequest)
+                      Future.successful(
+                        BadRequest(Json.toJson(ErrorResponseBuilder.failedXmlValidation))
+                          .as("application/json"))
                   }
```

Comment:
Do we have any plan's to pass the exact problem from the validation error (and all other errors) to the frontend, will this be done in this ticket or should we complete this at a later date. 

Again not necessary but it would help the frontend  and out logging of any problems with our xml and data modelling.